rskj-core/src/main/java/co/rsk/core/RskFactory.java

```diff
@@ -220,4 +244,46 @@ public PeerServer getPeerServer(SystemProperties config,
                                     EthereumChannelInitializerFactory ethereumChannelInitializerFactory) {
         return new PeerServerImpl(config, ethereumListener, ethereumChannelInitializerFactory);
     }
+
+    @Bean
+    public Wallet getWallet(RskSystemProperties config) {
+        if (!config.isWalletEnabled()) {
+            logger.info("Local wallet disabled");
+            return null;
+        }
+
+        logger.info("Local wallet enabled");
+        KeyValueDataSource ds = new LevelDbDataSource("wallet");
+        ds.init();
+        return new Wallet(ds);
+    }
+
+    @Bean
+    public PersonalModule getPersonalModuleWallet(Rsk rsk, Wallet wallet) {
+        if (wallet == null) {
+            return new PersonalModuleWalletDisabled();
```

Comment:
If PersonalModuleWalletDisabled reteurns a weird exception in all methods can't it be just null?